__tests__/buildx/bake.test.itg.ts

```diff
@@ -29,17 +29,26 @@ beforeEach(() => {
   jest.clearAllMocks();
 });
 
-maybe('parseDefinitions', () => {
+maybe('getDefinition', () => {
   // prettier-ignore
   test.each([
     [
-      ['https://github.com/docker/buildx.git#v0.10.4'],
+      'https://github.com/docker/buildx.git#v0.10.4',
       ['binaries-cross'],
       path.join(fixturesDir, 'bake-buildx-0.10.4-binaries-cross.json')
-    ]
-  ])('given %p', async (sources: string[], targets: string[], out: string) => {
+    ],
+    // TODO: enable this test when we are able to build from a remote definition on a private repo
+    // [
+    //   'https://github.com/docker/test-docker-action.git#remote-private',
+    //   ['default'],
+    //   path.join(fixturesDir, 'bake-buildx-0.10.4-binaries-cross.json')
+    // ]
```

Comment:
I got the following output when trying to build from a remote bake definition on a private repo:

```
$ docker buildx bake https://github.com/docker/test-docker-action.git#remote-private
#0 building with "default" instance using docker driver

#1 [internal] load git source https://github.com/docker/test-docker-action.git#remote-private
#1 0.225 fatal: could not read Username for 'https://github.com': terminal prompts disabled
#1 ERROR: failed to fetch remote https://github.com/docker/test-docker-action.git: git error: exit status 128
stderr:
fatal: could not read Username for 'https://github.com': terminal prompts disabled

------
 > [internal] load git source https://github.com/docker/test-docker-action.git#remote-private:
0.225 fatal: could not read Username for 'https://github.com': terminal prompts disabled
------
ERROR: couldn't find a bake definition
```

@tonistiigi Setting `GIT_AUTH_TOKEN` doesn't seem to work. Might need changes upstream in buildx.

Reply:
My bad I was setting it as build arg instead of secret :man_facepalming:

Reply:
Doesn't seem to work either with:

```
/usr/bin/docker buildx bake https://github.com/docker/test-docker-action.git#remote-private --set *.secrets=id=GIT_AUTH_TOKEN,src=/home/runner/work/_temp/docker-actions-toolkit-Rrz5jT/tmp-1910-aMBJe65r7Kf4 --print default
#0 building with "default" instance using docker driver
#1 [internal] load git source https://github.com/docker/test-docker-action.git#remote-private
#1 0.039 Initialized empty Git repository in /var/lib/docker/overlay2/l763ozu3v99vupy2gmketmi8t/diff/
#1 0.279 fatal: could not read Username for 'https://github.com/': terminal prompts disabled
```

which makes sense as loading remote bake definitions is done by a dedicated solve request, not during build: https://github.com/docker/buildx/blob/8abef5908705e49f7ba88ef8c957e1127b597a2a/bake/remote.go#L62

So would need to attach this secret to the session. UX wise I'm not sure what we could do. Using an override with `--set` is not correct so best guess is we need a new flag to bake cmd.

Reply:
moved last commit to for git auth token support in #288